x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/Div.java

```diff
@@ -76,8 +82,11 @@ static long processUnsignedLongs(long lhs, long rhs) {
         return asLongUnsigned(Long.divideUnsigned(asLongUnsigned(lhs), asLongUnsigned(rhs)));
     }
 
-    @Evaluator(extraName = "Doubles")
+    @Evaluator(extraName = "Doubles", warnExceptions = { ArithmeticException.class })
     static double processDoubles(double lhs, double rhs) {
+        if (rhs == 0) {
+            throw new ArithmeticException("Division by zero");
+        }
         return lhs / rhs;
```

Comment:
I think this does not cover all the edge cases if we care about infinities. Dividing something big by something small can still end up infinite.

Maybe in this case it's easier to just perform `lhs/rhs` and throw an `ArithmeticException` if the result is not finite?